File: Training/OOPFundamentals/Java/Exercises/after/05 Method Overlod/src/ShoppingCart.java
public class ShoppingCart {

    private float totalPrice;
    private int numberOfItems;

    public float getTotalPrice() {
        return totalPrice;
    }

    public int getNumberOfItems() {
        return numberOfItems;
    }

    public void addItems(float price, int numberOfItems) {
        this.totalPrice += price;
        this.numberOfItems += numberOfItems;
    }

    public void addItems(float price) {
        this.addItems(price, 1);
    }

    public void addItems(Item item) {
        float itemPrice = item.getPrice();
        this.addItems(itemPrice);
    }

    public void addItems(Item... list) {
        for(Item item : list) {
            this.addItems(item);
        }
    }

}

File: Training/OOPFundamentals/C#/Exercises/after/MethodOverload/ShoppingCart.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MethodOverload
{
    public class ShoppingCart
    {
        private float _totalPrice;
        private int _numberOfItems;

        public float TotalPrice { get
            {
                return _totalPrice;
            }
        }

        public int NumberOfItems { get
            {
                return _numberOfItems;
            }
        }

        //solution
        public void AddItems(float price, int numberOfItems)
        {
            _totalPrice += price;
            _numberOfItems += numberOfItems;
        }

        public void AddItems(float price)
        {
            _totalPrice += price;
            _numberOfItems++;
        }

        public void AddItems(params Item[] items)
        {
            foreach(Item item in items)
            {
                _totalPrice += item.Price;
            }
            _numberOfItems += items.Length;
        }
    }
}
